ui/pages/confirmations/components/confirm/info/typed-sign/permit-simulation/permit-simulation.tsx

```diff
@@ -1,118 +1,22 @@
-import PermitSimulationValueDisplay from './value-display/value-display';
-
-function extractTokenDetailsByPrimaryType(
-  message: Record<string, unknown>,
-  primaryType: PrimaryType,
-): object[] | unknown {
-  let tokenDetails;
-
-  switch (primaryType) {
-    case PrimaryType.PermitBatch:
-    case PrimaryType.PermitSingle:
-      tokenDetails = message?.details;
-      break;
-    case PrimaryType.PermitBatchTransferFrom:
-    case PrimaryType.PermitTransferFrom:
-      tokenDetails = message?.permitted;
-      break;
-    default:
-      break;
-  }
-
-  const isNonArrayObject = tokenDetails && !Array.isArray(tokenDetails);
 
-  return isNonArrayObject ? [tokenDetails] : tokenDetails;
-}
+import { SignatureRequestType } from '../../../../../types/confirm';
+import { useConfirmContext } from '../../../../../context/confirm';
+import { DefaultSimulation } from './default-simulation';
+import { DecodedSimulation } from './decoded-simulation';
 
 const PermitSimulation: React.FC<object> = () => {
-  const t = useI18nContext();
   const { currentConfirmation } = useConfirmContext<SignatureRequestType>();
-  const msgData = currentConfirmation.msgParams?.data;
-  const chainId = currentConfirmation.chainId as Hex;
-  const {
-    domain: { verifyingContract },
-    message,
-    message: { tokenId },
-    primaryType,
-  } = parseTypedDataMessage(msgData as string);
-  const isNFT = tokenId !== undefined;
-
-  const tokenDetails = extractTokenDetailsByPrimaryType(message, primaryType);
+  const { decodingLoading, decodingData } = currentConfirmation;
 
-  const TokenDetail = ({
-    token,
-    amount,
-    i,
-  }: {
-    token: Hex | string;
-    amount: number | string;
-    i: number;
-  }) => (
-    <PermitSimulationValueDisplay
-      key={`${token}-${i}`}
-      primaryType={primaryType}
-      tokenContract={token}
-      value={amount}
-      chainId={chainId}
-    />
-  );
-
-  const SpendingCapRow = (
-    <ConfirmInfoRow
-      label={t(isNFT ? 'simulationApproveHeading' : 'spendingCap')}
-    >
-      <Box style={{ marginLeft: 'auto', maxWidth: '100%' }}>
-        {Array.isArray(tokenDetails) ? (
-          <Box
-            display={Display.Flex}
-            flexDirection={FlexDirection.Column}
-            gap={2}
-          >
-            {tokenDetails.map(
-              (
-                { token, amount }: { token: string; amount: string },
-                i: number,
-              ) => (
-                <TokenDetail token={token} amount={amount} i={i} />
-              ),
-            )}
-          </Box>
-        ) : (
-          <PermitSimulationValueDisplay
-            tokenContract={verifyingContract}
-            value={message.value}
-            tokenId={message.tokenId}
-            chainId={chainId}
-          />
-        )}
-      </Box>
-    </ConfirmInfoRow>
-  );
+  if (
+    decodingData?.error ||
+    (decodingData === undefined && decodingLoading !== true)
+  ) {
+    return <DefaultSimulation />;
+  }
```

Comment:
Here we display decoding api changes if present and fall back to old simulation implementation otherwise.